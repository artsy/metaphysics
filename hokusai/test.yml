version: "2"
services:
  metaphysics:
    command: env NODE_OPTIONS=--max_old_space_size=4096 yarn ci
    extends:
      file: build.yml
      service: metaphysics
    environment:
      - MEMCACHED_URL=metaphysics-memcached:11211
      - CI_PULL_REQUEST=$CI_PULL_REQUEST
      - CIRCLE_PULL_REQUEST=$CIRCLE_PULL_REQUEST
      - CIRCLE_BUILD_NUM=$CIRCLE_BUILD_NUM
      - ENABLE_CONSIGNMENTS_STITCHING=true
    env_file: ../.env.test
    depends_on:
      - metaphysics-memcached
  metaphysics-memcached:
    image: memcached
    ports:
      - 11211:11211
