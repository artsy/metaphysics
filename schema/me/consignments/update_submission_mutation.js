import { mutationWithClientMutationId } from "graphql-relay"
import { SubmissionType } from "./submission"

export const config = {
  name: "UpdateSubmissionMutation",
  description: "Update a consigment using Convection",
  inputFields: {
    ...SubmissionType.getFields(),
  },
  outputFields: {
    submission: {
      type: SubmissionType,
      resolve: submission => submission,
    },
  },
  mutateAndGetPayload: (submission, request, { rootValue: { submissionUpdateLoader } }) => {
    if (!submissionUpdateLoader) return null
    return submissionUpdateLoader(submission.id, submission)
  },
}
export default mutationWithClientMutationId(config)
