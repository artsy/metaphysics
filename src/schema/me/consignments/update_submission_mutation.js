// @ts-check
import { mutationWithClientMutationId } from "graphql-relay"
import { SubmissionType } from "./submission"
import { omit } from "lodash"

export const config = {
  name: "UpdateSubmissionMutation",
  description: "Update a consigment using Convection",
  inputFields: {
    ...omit(SubmissionType.getFields(), ["__id", "_id", "artist"]),
  },
  outputFields: {
    consignment_submission: {
      type: SubmissionType,
      resolve: submission => submission,
    },
  },
  mutateAndGetPayload: (
    submission,
    _request,
    { rootValue: { submissionUpdateLoader } }
  ) => {
    if (!submissionUpdateLoader) return null
    return submissionUpdateLoader(submission.id, submission)
  },
}
export default mutationWithClientMutationId(config)
