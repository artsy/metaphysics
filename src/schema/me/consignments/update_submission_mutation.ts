import { mutationWithClientMutationId, MutationConfig } from "graphql-relay"
import { SubmissionType } from "./submission"
import { omit } from "lodash"
import { ResolverContext } from "types/graphql"

export const config: MutationConfig<any, any, ResolverContext> = {
  name: "UpdateSubmissionMutation",
  description: "Update a consignment using Convection",
  inputFields: {
    ...omit(SubmissionType.getFields(), ["__id", "_id", "artist"]),
  } as any,
  outputFields: {
    consignment_submission: {
      type: SubmissionType,
      resolve: submission => submission,
    },
  },
  mutateAndGetPayload: (submission, { submissionUpdateLoader }) => {
    if (!submissionUpdateLoader) return null
    return submissionUpdateLoader(submission.id, submission)
  },
}

export default mutationWithClientMutationId(config)
