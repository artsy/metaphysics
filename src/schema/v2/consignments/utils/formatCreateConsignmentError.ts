import {
  GraphQLInt,
  GraphQLNonNull,
  GraphQLObjectType,
  GraphQLString,
} from "graphql"
import { formatGravityError } from "lib/gravityErrorHandler"
import { ResolverContext } from "types/graphql"

export const ConsignmentInquiryMutationErrorName =
  "ConsignmentInquiryMutationError"

export const ConsignmentInquiryErrorType = new GraphQLObjectType<
  any,
  ResolverContext
>({
  name: ConsignmentInquiryMutationErrorName,
  fields: () => ({
    type: {
      type: GraphQLString,
    },
    message: {
      type: new GraphQLNonNull(GraphQLString),
    },
    error: {
      type: GraphQLString,
    },
    statusCode: {
      type: GraphQLInt,
    },
  }),
})

export const formatCreateConsignmentError = (error) => {
  return formatGravityError(error)
}
