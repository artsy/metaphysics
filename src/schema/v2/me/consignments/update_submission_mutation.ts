/**
 * FIXME: Deprecated. Remove once Convection stitching settles for a bit
 */

import { mutationWithClientMutationId, MutationConfig } from "graphql-relay"
import {
  SubmissionType,
  SubmissionCategoryAggregation,
  SubmissionDimensionAggregation,
  SubmissionStateAggregation,
} from "./submission"
import { ResolverContext } from "types/graphql"
import {
  GraphQLNonNull,
  GraphQLString,
  GraphQLBoolean,
  GraphQLInt,
  GraphQLID,
} from "graphql"

export const config: MutationConfig<any, any, ResolverContext> = {
  name: "UpdateSubmissionMutation",
  description: "Update a consignment using Convection",
  inputFields: {
    id: {
      description: "The GUID for the submission",
      type: new GraphQLNonNull(GraphQLString),
    },
    artistID: {
      description: "The gravity ID for an Artist",
      type: new GraphQLNonNull(GraphQLString),
    },
    authenticityCertificate: {
      description: "Does the artwork come with an certificate of authenticity?",
      type: GraphQLBoolean,
    },
    category: {
      description: "The set in which to put the work",
      type: SubmissionCategoryAggregation,
    },
    depth: {
      description: "The depth of the work",
      type: GraphQLString,
    },
    dimensionsMetric: {
      description: "A string, either CM or IN",
      type: SubmissionDimensionAggregation,
    },
    edition: {
      description: "Is the work a part of an edition",
      type: GraphQLBoolean,
    },
    editionNumber: {
      description: "The number of the individual work if in a set",
      type: GraphQLString,
    },
    editionSize: {
      description: "The whole size of the set of works",
      type: GraphQLInt,
    },
    height: {
      description: "The height of the work",
      type: GraphQLString,
    },
    locationCity: {
      description: "The city where the work currently resides",
      type: GraphQLString,
    },
    locationCountry: {
      description: "The country where the work currently resides",
      type: GraphQLString,
    },
    locationState: {
      description: "The state where the work currently resides",
      type: GraphQLString,
    },
    medium: {
      description: "The materials in which the work is created",
      type: GraphQLString,
    },
    provenance: {
      description: "The history of an work",
      type: GraphQLString,
    },
    signature: {
      description: "Is this work signed?",
      type: GraphQLBoolean,
    },
    title: {
      description: "The name of the work",
      type: GraphQLString,
    },
    state: {
      description: "The internal state of the work, e.g. draft/submitted",
      type: SubmissionStateAggregation,
    },
    width: {
      description: "The width of the work",
      type: GraphQLString,
    },
    year: {
      description: "The year the work was created",
      type: GraphQLString,
    },
    userID: {
      description: "The user who submitted the work",
      type: GraphQLID,
    },
  } /*
  FIXME: Generated by the snake_case to camelCase codemod.
         Either use this to fix inputs and/or remove this comment.
  {
    const {
      artistID,
      authenticityCertificate,
      dimensionsMetric,
      editionNumber,
      editionSize,
      locationCity,
      locationCountry,
      locationState,
      userID,
      ..._newFields
    } = newFields;
    const oldFields = {
      artistID: artist_id,
      authenticityCertificate: authenticity_certificate,
      dimensionsMetric: dimensions_metric,
      editionNumber: edition_number,
      editionSize: edition_size,
      locationCity: location_city,
      locationCountry: location_country,
      locationState: location_state,
      userID: user_id,
      ..._newFields
    };
  }
  */,
  outputFields: {
    consignmentSubmission: {
      type: SubmissionType,
      resolve: submission => submission,
    },
  },
  mutateAndGetPayload: (
    {
      artistID,
      authenticityCertificate,
      dimensionsMetric,
      editionNumber,
      editionSize,
      locationCity,
      locationCountry,
      locationState,
      userID,
      ..._submission
    },
    { submissionUpdateLoader }
  ) => {
    const submission: any = {
      artist_id: artistID,
      authenticity_certificate: authenticityCertificate,
      dimensions_metric: dimensionsMetric,
      edition_number: editionNumber,
      edition_size: editionSize,
      location_city: locationCity,
      location_country: locationCountry,
      location_state: locationState,
      user_id: userID,
      ..._submission,
    }
    if (!submissionUpdateLoader) return null
    return submissionUpdateLoader(submission.id, submission)
  },
}

export default mutationWithClientMutationId(config)
