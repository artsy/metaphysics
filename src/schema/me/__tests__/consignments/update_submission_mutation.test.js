import { runAuthenticatedQuery } from "test/utils"
import { config as updateSubmissionMutation } from "schema/me/consignments/update_submission_mutation.js"
import gql from "test/gql"

describe("UpdateSubmissionMutation", () => {
  it("includes the id param", () => {
    const mutation = updateSubmissionMutation
    expect(Object.keys(mutation.inputFields)).toContain("id")
  })

  it("includes the state param", () => {
    const mutation = updateSubmissionMutation
    expect(Object.keys(mutation.inputFields)).toContain("state")
  })

  it("updates a submission and returns its new data payload", () => {
    const mutation = gql`
      mutation {
        updateConsignmentSubmission(
          input: {
            id: "108"
            artist_id: "andy-warhol"
            depth: "123"
            clientMutationId: "123123"
          }
        ) {
          clientMutationId
          consignment_submission {
            depth
          }
        }
      }
    `

    const rootValue = {
      submissionUpdateLoader: () =>
        Promise.resolve({
          id: "106",
          artist_id: "andy-warhol",
          authenticity_certificate: true,
          depth: "1000",
        }),
    }

    expect.assertions(1)
    return runAuthenticatedQuery(mutation, rootValue).then(data => {
      expect(data).toMatchSnapshot()
    })
  })
})
