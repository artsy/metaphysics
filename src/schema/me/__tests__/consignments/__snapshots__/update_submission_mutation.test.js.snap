// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateSubmissionMutation updates a submission and returns its new data payload 1`] = `
Object {
  "updateConsignmentSubmission": Object {
    "clientMutationId": "123123",
    "consignment_submission": Object {
      "depth": "1000",
    },
  },
}
`;
